### PR TITLE
feat(swift): handle generics

### PR DESCRIPTION
**Pull request by Fluf22**
## 🧭 What and Why

🎟 JIRA Ticket: [DI-1750]

### Changes included:

- use generic types when possible

[DI-1750]: https://algolia.atlassian.net/browse/DI-1750?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by shortcuts — COMMENTED** [review]
Huuuge

File: clients/algoliasearch-client-swift/Sources/Search/Extra/SearchClientExtension.swift [hunk 1]
```diff
@@ -322,6 +322,15 @@ public extension SearchClient {
             }
     }
 
+    func searchForHits<T: Codable>(
```
[the remark this answers is not in the document]
Comment (Fluf22) [reply clients/algoliasearch-client-swift/Sources/Search/Extra/SearchClientExtension.swift:332]: Yes, you're totally right, I forgot about that

**Comment by Fluf22** [comment]
Wow, everything is green, there must be something wrong with the CI

File: generators/src/main/java/com/algolia/codegen/AlgoliaSwiftGenerator.java [hunk 2]
```diff
@@ -281,64 +283,82 @@ private String typeToName(String content) {
...
+          modelHasEscapedPropertyName = true;
+        }
+
+        // Check if the property is null encodable
+        if (
+          property.vendorExtensions.containsKey("x-null-encodable") &&
+          property.vendorExtensions.get("x-null-encodable").toString().equals("true")
+        ) {
+          // Set null encodable default value
+          if (property.defaultValue != null && !property.defaultValue.equals("null")) {
```
Comment (millotp) [inline generators/src/main/java/com/algolia/codegen/AlgoliaSwiftGenerator.java:342]:
```suggestion
          if ("null".equals(property.defaultValue)) {
```

**Review by millotp — APPROVED** [review]
looks perfect gg ! small nit

**Pull request merged**